src/DependencyInjection/FOSHttpCacheExtension.php

```diff
@@ -131,6 +134,22 @@ public function load(array $configs, ContainerBuilder $container)
         }
     }
 
+    private function loadCacheable(ContainerBuilder $container, array $config)
+    {
+        $container->setParameter(
+            $this->getAlias().'.cacheable.response.additional_status',
+            $config['response']['additional_status']
+        );
+
+        $definition = $container->getDefinition($this->getAlias().'.response_matcher.cacheable');
+
+        // Change CacheableResponseMatcher to ExpressionResponseMatcher
+        if ($config['response']['expression']) {
+            $definition->setClass(ExpressionResponseMatcher::class)
+                ->setArguments([$config['response']['expression']]);
```

Comment:
This is a bit odd, but it works: when an expression is configured, switch to another RuleMatcher, which is dedicated to expressions.